src/shrinkage.jl

```diff
@@ -0,0 +1,84 @@
+"""
+    CoefByGroup
+
+Coefficients for groups of observations defined by levels of a grouping factor
+
+Fields include:
+- `cnames`: `Vector{String}` of the column (or coefficient) names
+- `levels`: `Vector` of the levels of the grouping factor.  Usually `String`s but not necessarily.
+- `fixed`: `AbstractVector{T}` of the global OLS estimates for the coefficients in `cnames` only.
+- `condmodes`: `AbstractMatrix{T}` of size `length(levels) x length(cnames)` of the conditional means/modes for the random effects
+- `grpest`: similar to `condmodes` but allowing for `Missing` values, giving the within group OLS estimates of the coefficients.
+"""
+struct CoefByGroup{T<:AbstractFloat}
+    cnames::Vector{String}
+    levels::Vector
+    fixed::AbstractVector{T}
+    condmodes::AbstractMatrix{T}
+    grpest::AbstractMatrix{Union{Missing,T}}
+end
```

Comment:
Should the blocking variable be tracked as part of this struct?

Looking below, it's usually wrapped in a NamedTuple with the blocking variable, so I don't know how useful that would be.

Reply:
I did at one time include the blocking factor name then decided to drop it as a field and use the `NamedTuple`.  I don't have strong opinions either way.